File: app/src/main/java/com/oxapps/materialcountdown/Event.java
package com.oxapps.materialcountdown;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by flynn on 21/11/15.
 */
public class Event implements Parcelable {
    protected static final String CATEGORY = "category";
    private int id;
    private String name;
    private String description;
    private long endTime;
    private Category category;

    public Event(String name, String description, long endTime, Category category) {
        this.name = name;
        this.description = description;
        this.endTime = endTime;
        this.category = category;
    }

    public Event(int id, String name, String description, long endTime, Category category) {
        this(name, description, endTime, category);
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public long getEndTime() {
        return endTime;
    }

    public long getRemainingTime() {
        return endTime - System.currentTimeMillis();
    }

    public Category getCategory() {
        return category;
    }

    public String getDescription() { return description;}

    protected Event(Parcel in) {
        id = in.readInt();
        name = in.readString();
        description = in.readString();
        endTime = in.readLong();
        category = (Category) in.readValue(Category.class.getClassLoader());
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
        dest.writeString(name);
        dest.writeString(description);
        dest.writeLong(endTime);
        dest.writeValue(category);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<Event> CREATOR = new Parcelable.Creator<Event>() {
        @Override
        public Event createFromParcel(Parcel in) {
            return new Event(in);
        }

        @Override
        public Event[] newArray(int size) {
            return new Event[size];
        }
    };

    public int getId() {
        return id;
    }
}

File: app/src/main/java/com/oxapps/materialcountdown/db/Event.kt
package com.oxapps.materialcountdown.db

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.os.Parcelable
import com.oxapps.materialcountdown.db.util.CATEGORY
import com.oxapps.materialcountdown.db.util.DESC
import com.oxapps.materialcountdown.db.util.END
import com.oxapps.materialcountdown.db.util.NAME
import com.oxapps.materialcountdown.model.Category
import kotlinx.android.parcel.Parcelize

@Entity(tableName = "events")
@Parcelize
data class Event(
    @PrimaryKey(autoGenerate = true) var id: Long?,
    @ColumnInfo(name = NAME) var name: String,
    @ColumnInfo(name = DESC) var description: String?,
    @ColumnInfo(name = END) var endTime: Long,
    @ColumnInfo(name = CATEGORY) var category: Category
): Parcelable {


    fun getRemainingTime(): Long {
        return endTime - System.currentTimeMillis()
    }
}